util/dynamicConfig.js

```diff
@@ -1,29 +1,62 @@
+/* eslint-disable no-param-reassign */
 const get = require("get-value");
 const unset = require("unset-value");
 
-function getDynamicConfig(event) {
-  const { Config } = event.destination;
-  if (Config) {
-    Object.keys(Config).forEach(field => {
-      let value = Config[field].toString().trim();
-      if (value.startsWith("{{") && value.endsWith("}}")) {
-        value = value.replace("{{", "");
-        value = value.replace("}}", "");
-        if (value.includes("||")) {
-          const path = value.split("||")[0].trim();
-          const getFieldVal = get(event, path);
-          if (getFieldVal) {
-            Config[field] = getFieldVal;
-            unset(event, path);
-          } else {
-            Config[field] = JSON.parse(value.split("||")[1].trim()).toString();
-          }
-        }
+function isValidPattern(val) {
+  // this rgegex checks for pattern  "only spaces {{ path || defaultvalue }}  only spaces" .
+  //  " {{message.traits.key  ||   \"email\" }} "
+  //  " {{ message.traits.key || 1233 }} "
+  const re = /^\s*\{\{\s*(?<path>[a-zA-Z_]([a-zA-Z0-9_]*\.[a-zA-Z_][a-zA-Z0-9_]*)+)+\s*\|\|\s*"?(?<defaultValue>.*)"?\s*\}\}\s*$/;
+  return re.test(String(val));
+}
+
+function getDynamicConfigValue(event, value) {
+  value = value.replace("{{", "").replace("}}", "");
+  if (value.includes("||")) {
+    const path = value.split("||")[0].trim();
+    const fieldVal = get(event, path);
+    if (fieldVal) {
+      value = fieldVal;
+      unset(event, path);
+    } else {
+      value = JSON.parse(value.split("||")[1].trim());
+    }
+  }
+  return value;
+}
+
+// eslint-disable-next-line consistent-return
+function configureVal(value, event) {
+  if (value) {
+    if (Array.isArray(value)) {
+      value.forEach((key, index) => {
+        value[index] = configureVal(key, event);
+      });
+    } else if (typeof value === "object") {
+      Object.keys(value).forEach(obj => {
+        value[obj] = configureVal(value[obj], event);
+      });
+    } else if (
+      typeof value !== "function" &&
+      typeof value !== "symbol" &&
+      typeof value !== "boolean"
+    ) {
+      const val = value.toString().trim();
+      if (isValidPattern(val)) {
+        value = getDynamicConfigValue(event, val);
       }
-    });
+    }
+    return value;
```

Comment:
This is not required.

Reply:
If value parameter is a boolean then it does logical operation in the if condition therefore I gave a return condition here. We can put it inside a condition.

Reply:
It would still be not required as,
- `true` - boolean data type check passes and would return
- `false` - the check would fail and would return

Reply:
yes in case of fail it would return but without any value thus leading it to be `undefined`

Reply:
It would return `value` (false), right?

Reply:
oh sorry. you are talking about return statement inside the if condition.

Reply:
updated